### PR TITLE
New door icons, bugfix "keylist in normal", bugfix "chest game death to pyramid"

### PR DESCRIPTION
**Pull request by SmolBig**
Found an existing keysanity config bit written to a common location (in Patch.cs) so I removed the config bit in the asm project and just did a bit test on the existing bit.

### CODE REVIEW

**Comment by SmolBig** [comment]
Added bugfix for non keysanity map showing key list.

**Comment by SmolBig** [comment]
Added icons for new doors.

**Comment by SmolBig** [comment]
added fix for "death to bombs in chest game resurrects on pyramid" bug

File: src/sm/keycards.asm [hunk 1]
```diff
@@ -121,7 +121,11 @@ keycard_update_layer3:
     dex #2
     bpl -
 
+    LDA $80FF52 ; load config bits
```
Comment (tewtal) [inline src/sm/keycards.asm:124]: I'm not sure if we should be using the seed config field for this. That field has very limited amount of data and can not properly represent all the configuration that might be needed for things. I think the previous solution with a "config_keysanity" is better here, or a specific flag for these keycard markers.

File: src/sm/map_icons.asm [hunk 2]
```diff
@@ -6,12 +6,11 @@ Return_From_Hijack:
 org $C2FB80 ; Beginning of a large block of free space
 base $82FB80
 Map_Labels_Hijack:
-PHA ; getting the keysanity flag in A, so let's preserve it first
-LDA config_keysanity ; check to see if this is keysanity
+LDA $80FF52 ; load config bits
```
Comment (tewtal) [inline src/sm/map_icons.asm:9]: Similar thing here as in keycards.asm, not a big fan of using the seed identifier fields for this.

File: src/z3/randomizer/darkworldspawn.asm [hunk 3]
```diff
@@ -48,7 +48,10 @@ SetDeathWorldChecked:
 	LDA $1B : BEQ + ; skip this for indoors
 		LDA $040C : CMP #$FF : BNE .done ; unless it's a cave
 
-		LDA $A0 : BNE + : LDA GanonPyramidRespawn : BEQ + ; check if we died in ganon's room and pyramid respawn is enabled
+    ; check if we died in ganon's room and pyramid respawn is enabled
+		LDA $A0 : BNE +
+    LDA $A1 : BNE + 
+    LDA GanonPyramidRespawn : BEQ +
```
Comment (tewtal) [inline src/z3/randomizer/darkworldspawn.asm:54]: While this is a fix we want, it should probably be in its own PR so we can merge it directly into something like a V11.1.1 hotfix release or similar without getting any keysanity related changes with it.

**Comment by RebelusQuo** [comment]
This PR has become a big mess of dissociated changes. My vote is for close and start over. We want the contribution, but it needs a good packaging as well.

**Comment by SmolBig** [comment]
Restarting.
